byterun/alloc.c

```diff
@@ -214,7 +218,11 @@ CAMLprim value caml_alloc_dummy_function(value size,value arity)
 /* [size] is a [value] representing number of floats. */
 CAMLprim value caml_alloc_dummy_float (value size)
 {
+#ifdef FLAT_FLOAT_ARRAY
```

Comment:
This does not look correct to me.  caml_alloc_dummy_float is used to compile rec-bindings, in two cases: (i) unboxed float arrays; (ii) unboxed float records.  AFAICT, (i) cannot happen any more (there are not Floatarray litterals) but (ii) can still happen.  So I think caml_alloc_dummy_float should be left unchanged.